client/deployments/client.go

```diff
@@ -49,15 +88,92 @@ func NewClient(url string, skipVerify bool) *Client {
 
 	return &Client{
 		url:               url,
-		artifactUploadUrl: JoinURL(url, artifactUploadUrl),
+		artifactUploadURL: JoinURL(url, artifactUploadURL),
+		artifactsListURL:  JoinURL(url, artifactsListURL),
 		client: &http.Client{
 			Transport: tr,
 		},
 	}
 }
 
-func (c *Client) UploadArtifact(description, artifactPath, tokenPath string, noProgress bool) error {
+func (c *Client) ListArtifacts(tokenPath string, detailLevel int) error {
+	token, err := ioutil.ReadFile(tokenPath)
```

Comment:
Question:
I don't think this captures this error completely. I think that the cached token can be present, but you still get `Unathorized`. Which would lead to the HTTP request returning `401` below. So I think an explicit login is required, as compared to this?

Reply:
In fact i like the idea of having a file containing the token for auth lying around somewhere. I already thought of a way to put the token in the `.mender-clirc` file instead of putting user and pw in it. 
But to make this feature really useful there must be a way to create tokens in the cloud app with specific rights and a way to revoke those tokens. So basically i'd preferr trying to read the token from a file (for the moment from cached file, later from `.mender-clirc`). Currently i'd stick with the solution and plan to implement the feature mentioned above. 

With this implementation when the token is invalid an error is reported:

```
./mender-cli artifacts list
Using configuration file: /home/armin/.mender-clirc
FAILURE: Get /deployments/artifacts request failed: Get "https://hosted.mender.io/api/management/v1/deployments/artifacts": net/http: invalid header field value "Bearer eyJhbGciOiJSUzI1.........RKlU\n" for key Authorization
```

Reply:
> In fact i like the idea of having a file containing the token for auth lying around somewhere. I already thought of a way to put the token in the `.mender-clirc` file instead of putting user and pw in it.

Yes, I agree.

> But to make this feature really useful there must be a way to create tokens in the cloud app with specific rights and a way to revoke those tokens. So basically i'd preferr trying to read the token from a file (for the moment from cached file, later from `.mender-clirc`). Currently i'd stick with the solution and plan to implement the feature mentioned above.
> 
> With this implementation when the token is invalid an error is reported:
> 
> ```
> ./mender-cli artifacts list
> Using configuration file: /home/armin/.mender-clirc
> FAILURE: Get /deployments/artifacts request failed: Get "https://hosted.mender.io/api/management/v1/deployments/artifacts": net/http: invalid header field value "Bearer eyJhbGciOiJSUzI1.........RKlU\n" for key Authorization
> ```

This I'm not sure of. Should this not return `401` ?

Reply:
Yes, you are right. I modified the authtoken file manually, Unfortunately this is not the way to go to invalidate the token... Anyway, is it okay for you to stick with the current solution for now and build the more advanced token handling in the `.mender-clirc` file?

Reply:
Yes, that is fine with me for now :)